src/main/java/core/basesyntax/service/impl/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.Fruit;
+
+public class PurchaseOperation implements OperationHandler {
+    private final FruitStorageDao fruitStorageDao;
+
+    public PurchaseOperation(FruitStorageDao fruitStorageDao) {
+        this.fruitStorageDao = fruitStorageDao;
+    }
+
+    @Override
+    public boolean doOperation(Fruit fruit, int quantity) {
+        if (fruit == null) {
+            throw new RuntimeException("Fruit can't be null");
+        }
+        if (quantity < MIN_FRUIT_QUANTITY) {
+            throw new RuntimeException("Quantity " + quantity + " can't be negative");
```

Comment:
It looks like MIN_FRUIT_QUANTITY might be an undeclared or undefined variable. Make sure to declare it with a proper access modifier and initialize it with a value that represents the minimum allowed quantity. This could be a constant within the class if it's a fixed value.